packages/functions/src/jobs/CronJobs.ts

```diff
@@ -52,12 +56,16 @@ export const SendConfirmationReminders = async (
 
   const donorIds = appointments.map((appointment) => appointment.donorId);
 
-  const donorsInAppointments = await getDonors(_.uniq(donorIds));
+  const donorsInAppointments: { [donorId: string]: DbDonor } = (
+    await getDonors(_.uniq(donorIds))
+  ).reduce((obj, donor: DbDonor) => {
+    return { ...obj, [donor.id]: donor };
+  }, {});
+
+  delete donorsInAppointments[MANUAL_DONOR_ID];
```

Comment:
Use `donorIds.filter(id => id  !=== MANUAL_DONOR_ID)`

Reply:
I added this code for improving time complexity in line 68..

Reply:
however maybe it worth adding a comment of what those lines do..

Reply:
It is great, just don't use `delete`. 
Also, there's no need for `reduce`.

`.filter` will get you what you want, no?

Reply:
In line 68, I want to get the DBDonor object from the donorId that I get from the appointment.
fot that, I need a dictionary like object that the key is the donorId and the value is the donor object, in that way I can get the donor object from the donorId in O(1) complexity.
so what I need to do is to build this dictionary..
what I have for building this dictionary is an array of DBDonor objects (from getDonor(...)) and inside eatch donor object there is a donorId that should turn to the key of the item in the dictionary.
filter will return an array, and that is not what I need. with reduce I can build this dictionary in an easy way.

Reply:
about the delete, you are probebly write becouse the getDonors() function will not return items with id of MANUAL_DONOR_ID..

Reply:
So why not use `map`?

```
const donorsInAppointments = await getDonors(_.uniq(donorIds));
const donorsMap : { [donorId: string]: DbDonor};
donorsInAppointments.map(donor => donorsMap[donor.id]=donor } 
```